src/Adapter/Product/AdminProductWrapper.php

```diff
@@ -83,20 +84,27 @@ class AdminProductWrapper
      */
     private $floatParser;
 
+    /**
+     * @var HookDispatcherInterface
+     */
+    private $hookDispatcher;
+
     /**
      * Constructor : Inject Symfony\Component\Translation Translator.
      *
      * @param object $translator
      * @param array $employeeAssociatedShops
      * @param Locale $locale
      * @param FloatParser|null $floatParser
+     * @param HookDispatcher $hookDispatcher
      */
-    public function __construct($translator, array $employeeAssociatedShops, Locale $locale, FloatParser $floatParser = null)
+    public function __construct($translator, array $employeeAssociatedShops, Locale $locale, FloatParser $floatParser = null, HookDispatcherInterface $hookDispatcher)
```

Comment:
It's a BC break, can you explain it in your PR description please?

Reply:
Oh yes, didn't notice this.

Should I leave this BC or should I implement the hook call the "old" way ?

Reply:
I prefer the BC than using singleton :)

Reply:
OK, then I will also change all the other hook call in this file so it will be cleaner